### PR TITLE
CPLP-1917 add company-identifiers to administration company-details response

### PR DESCRIPTION
**Pull request by ntruchsess**
implements https://jira.catena-x.net/browse/CPLP-1917

### CODE REVIEW

File: src/mailing/Mailing.Template/TemplateManager.cs [hunk 1]
```diff
@@ -31,6 +31,8 @@ public class TemplateManager : ITemplateManager
 {
     private readonly TemplateSettings _settings;
 
+    private static readonly Regex _templateMatcherExpression = new Regex(@"\{(\w+)\}", RegexOptions.None, TimeSpan.FromSeconds(1)); // to replace any text surrounded by { and }
```
Comment (ntruchsess) [inline src/mailing/Mailing.Template/TemplateManager.cs:34]: this is a 'drive-by' fix of a sonarcloud security finding.

File: src/administration/Administration.Service/BusinessLogic/IdentityProviderBusinessLogic.cs [hunk 2]
```diff
@@ -42,6 +42,8 @@ public class IdentityProviderBusinessLogic : IIdentityProviderBusinessLogic
     private readonly IProvisioningManager _provisioningManager;
     private readonly IdentityProviderSettings _settings;
 
+    private static readonly Regex _displayNameValidationExpression = new Regex(@"^[a-zA-Z0-9\!\?\@\&\#\'\x22\(\)_\-\=\/\*\.\,\;\: ]+$", RegexOptions.None, TimeSpan.FromSeconds(1));
```
Comment (ntruchsess) [inline src/administration/Administration.Service/BusinessLogic/IdentityProviderBusinessLogic.cs:45]: this is a 'drive-by' fix of a sonarcloud security finding.

File: src/portalbackend/PortalBackend.DBAccess/Models/CompanyAddressIdentifierData.cs [hunk 3]
```diff
@@ -18,36 +18,27 @@
...
-    public string? Streetnumber { get; set; }
-
-    [JsonPropertyName("zipCode")]
-    public string? Zipcode { get; set; }
-
-    [JsonPropertyName("countryDe")]
-    public string? CountryDe { get; set; }
-    
-}
+public record CompanyAddressIdentifierData(
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.DBAccess/Models/CompanyAddressIdentifierData.cs:26]: I don't see the `CountryDe` property in the old repository code, therefore I assume that the Property was removed on purpose. Since the property is no longer included in the response, this could result in a breaking change for the frontend

**Review by Phil91 — APPROVED** [review]
Just a small question, overall looks good

File: src/portalbackend/PortalBackend.DBAccess/Models/CompanyAddressIdentifierData.cs (hunk 3 above)
Comment (ntruchsess) [reply src/portalbackend/PortalBackend.DBAccess/Models/CompanyAddressIdentifierData.cs:26]: The Model previously was shared with other endpoints which I have resolved recently. As the model now is not used by other endpoints any more I removed all redundant properties. (CountryDe was not used by this endpoint before this change).
